ev3dev/core.py

```diff
@@ -2033,12 +2035,26 @@ def mode(self):
         Returns the current mode. Writing one of the values returned by `modes`
         sets the sensor to that mode.
         """
-        self._mode, value = self.get_attr_string(self._mode, 'mode')
-        return value
+        if self._mode_value is None:
+            self._mode, value = self.get_attr_string(self._mode, 'mode')
+            self._mode_value = value
+            return value
+        else:
+            return self._mode_value
 
     @mode.setter
     def mode(self, value):
+        """
+        This will be called often so we cache the current mode in
+        self._mode_value to avoid some file IO.
+        """
+
+        # If 'mode' is already set to 'value' there is nothing to do.
+        if value == self._mode_value:
```

Comment:
Do we want a way to bypass this cache logic? Previously, the sysfs accessors were always guaranteed to perform a write operation, whereas now there's no way to handle the case where you intentionally want multiple processes interfacing with the same device. This is a result of the fact that caching was moved down a level from the friendly functions to the getters/setters.

Reply:
Part of me is wondering if we need it at all.  We didn't have it before and I've never noticed any perf issues with setting the modes.  We aren't setting the mode any more often with the deprecation of auto_mode (assuming nobody ever set this to false).  Maybe we just chop the `mode_value`/caching part of this patch?

Reply:
@WasabiFan what say you on the `mode_value`...keep or chop?  I'm leaning towards chop and address it as part of #365

Reply:
Agreed. I don't like the inconsistency this introduces, especially in light of cases such as the ultrasonic sensor where the only way to send a ping is to set the mode again. We should cut it from here and address it elsewhere.